### DISCUSSION TITLE
Is there any way to get the s3 bucket name of Supabase Storage

### DISCUSSION THREAD

**Discussion by hacking-impossible** [supabase/supabase#26806] [Questions]
Hello.
I am using the Cloudmailin to catch the email and upload email attachment to Supabase Storage.
But cloudmailin support only upload files to the AWS S3 and Google Cloud Storage.
To set the cloudmailin, I need the s3 bucket name, but I can't get it from Supabase Settings.
If anyone experienced in this field, please let me know as soon as possible.

**Comment by GaryAustin1** [comment]
There is not a real individual bucket behind Supabase storage.  A bucket is shared with lots of instances.
That said, there is a new s3 compatible API.  I do not know if it has what you are looking for but all the info is in the storage docs.
https://supabase.com/docs/guides/storage/s3/authentication?queryGroups=language&language=credentials

**Comment by nref** [comment]
TagSpaces also requires a bucket name. Landed here searching "supabase get bucket name"

**Reply by GaryAustin1** [reply]
As mentioned there is now an s3 compatible interface.  https://supabase.com/docs/guides/storage/s3/compatibility

**Reply by GaryAustin1** [reply]
If the new s3 interface does not give you the access as an s3 compatible bucket then there is no way.  There is no real bucket per instance in Supabase.  They share an s3 bucket with many instances.

**Reply by nref** [reply]
Thanks Gary

**Comment by nref** [comment]
Thanks for the info. That could be useful for others who land here. The fact that I am trying to use an S3 client with Supabase implies my awareness of the S3-compatible interface. The problem I have is that Supabase S3 is incompatible with many clients that were written with only the real S3 in mind, in particular those which require an explicit bucket name.

**Comment by nref** [comment]
If you ask "why" a couple times, the answer is my Supabase project is over its storage limit, and want to figure out what is using so much space using common S3 tools, and potentially delete those objects. I'd also like to sync them to local storage before doing so. At the moment I don't have any tools to perform those functions with Supabase storage.

**Comment by nref** [comment]
I was able to accomplish what I needed with [Rclone](https://rclone.org).

```
# Follow the interactive prompts. 
# When asked, provide the endpoint e.g. https://<project>.supabase.co/storage/v1/s3, 
# access key ID, and secret access key. I named the remote "supabase".
rclone config 
rclone ls supabase:/
rclone sync supabase:/ ~/some_folder # download all files
rclone delete supabase:/path/to/big-object
```

**Reply by Schotsl** [reply]
Could you share your rclone config? Of course, without the keys or endpoints, but just the options? I keep getting this error:

```NOTICE: Failed to ls: operation error S3: ListBuckets, https response error StatusCode: 403, RequestID: , HostID: , api error SignatureDoesNotMatch: The request signature we calculated does not match the signature you provided. Check your key and signing method.``` and I can't seem to fix it for the life of me, thank in advance!

I can't seem to fix it no matter what I try. Thanks in advance!

**Reply by nref** [reply]
Sure, here it is.

```
> rclone config dump
{
    "supabase": {
        "access_key_id": "<redacted>",
        "acl": "private",
        "endpoint": "https://<redacted>.supabase.co/storage/v1/s3",
        "provider": "Other",
        "region": "us-east-1",
        "secret_access_key": "<redacted>",
        "type": "s3"
    }
}
```

**Reply by Schotsl** [reply]
Thanks, @nref, but even with your config (using my keys and endpoint, of course), I keep getting the following error: ```2024/12/09 18:02:19 NOTICE: Failed to ls: operation error S3: ListBuckets, https response error StatusCode: 403, RequestID: , HostID: , api error SignatureDoesNotMatch: The request signature we calculated does not match the signature you provided. Check your key and signing method.```. 

Have you made any specific changes in Supabase to enable the connection? Also, could you run rclone -V so I can test with the same version? Thanks for your help!

**Reply by nref** [reply]
Sorry for the trouble you're having. Here's my rclone version:

```
> rclone -V
rclone v1.67.0
- os/version: Microsoft Windows 11 Pro 24H2 (64 bit)
- os/kernel: 10.0.26100.1742 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.22.4
- go/linking: static
- go/tags: cmount
```

As far as I remember and can tell, I haven't made any changes to Supabase. Here's my Supabase S3 configuration.

![image](https://github.com/user-attachments/assets/0f642c28-96d0-4615-92d9-010bb819c000)

I had a `SignatureDoesNotMatch` error with the real S3 once. In that case, I was using the wrong access key ID. https://stackoverflow.com/questions/37903462/aws-cli-listing-s3-buckets-gives-signaturedoesnotmatch-error-using-iam-user-cred

**Reply by Schotsl** [reply]
I figured it out! The problem is the rclone version. I'm running v1.68.0, while you're running v1.67.0. The major difference is the switch to AWS SDK v2 and the deprecation of v1. I'm guessing that Supabase is still running on v1, since using an older version like v1.67.0 works fine. There might be some configuration variable that can fix this, but I'm unsure which one. For now, I'm just running an older rclone version while waiting for Supabase to switch to v2. Thanks so much for your help @nref!

**Reply by nref** [reply]
You're welcome and I'm glad you're all set! I'm sure this will help others too.

**Reply by nref** [reply]
I wanted to also share that while rclone still works well, in the meantime I have also discovered the MinIO client. Here's a guide on how to use it.

https://brettweir.com/blog/hosting-static-sites-cloudflare-r2-minio-client/

**Reply by davidNbreslauer** [reply]
Seconded on the version issue for anyone else out there. Had the same problem just using rclone to transfer files from GDrive to Supabase (S3).

**Reply by CharlBest** [reply]
Same issue here.
I was using the AWS CLI version 2.24.20 (currently the latest) and all my images got corrupted when I upload them. Full file size would be 12kb but then uploaded verison is 10.5kb for example. Tried everything like different file types, binary upload, explicit mime type setting, disabled encryption  but couldn't get it work. Moved to rclone (version 1.69.1) and had the same issue as above. Now using 1.67.0 and all works good. Thanks for the help.

**Reply by nref** [reply]
Thanks @Schotsl for saving me a lot of trouble. Homebrew updated some time between December and now, and of course I forgot about this thread. Today the newer rclone choked with the same `SignatureDoesNotMatch` error. When I searched the web I was led right back here to the solution. 😄
